linkml/generators/erdiagramgen.py

```diff
@@ -0,0 +1,307 @@
+    def __str__(self):
+        return f'{self.left_cardinality}{self.identifying}{self.right_cardinality}'
+
+
+class Entity(pydantic.BaseModel):
+    """ Entity in an ER diagram."""
+    name: str
+    attributes: List[Attribute] = []
+
+    def __str__(self):
+        attrs = '\n'.join([str(a) for a in self.attributes])
+        return f'{self.name} {{\n{attrs}\n}}'
+
+
+class Relationship(pydantic.BaseModel):
+    """ Relationship in an ER diagram."""
+    first_entity: str
+    relationship_type: RelationshipType = None
+    second_entity: str = None
+    relationship_label: str = None
+
+    def __str__(self):
+        return f'{self.first_entity} {self.relationship_type} {self.second_entity} : "{self.relationship_label}"'
+
+
+class ERDiagram(pydantic.BaseModel):
+    """Represents an Diagram of Entities and Relationships"""
+    entities: List[Entity] = []
+    relationships: List[Relationship] = []
+
+    def __str__(self):
+        ents = '\n'.join([str(e) for e in self.entities])
+        rels = '\n'.join([str(r) for r in self.relationships])
+        return f'erDiagram\n{ents}\n\n{rels}\n'
+
+
+@dataclass
+class ERDiagramGenerator(Generator):
+    """
+    A generator for serializing schemas as Entity-Relationship diagrams.
+
+    Currently this generates diagrams in mermaid syntax, but in future
+    this could easily be extended to have for example a direct SVG or PNG
+    generator using PyGraphViz, similar to erdantic.
+    """
+
+    # ClassVars
+    generatorname = os.path.basename(__file__)
+    generatorversion = "0.0.1"
+    valid_formats = ["markdown", "mermaid"]
+    uses_schemaloader = False
+    requires_metamodel = False
+
+    structural: bool = True
+    """If True, then only the tree_root and entities reachable from the root are drawn"""
+
+    exclude_attributes: bool = False
+    """If True, do not include attributes in entities"""
+
+    genmeta: bool = field(default_factory=lambda: False)
+    gen_classvars: bool = field(default_factory=lambda: True)
+    gen_slots: bool = field(default_factory=lambda: True)
+    no_types_dir: bool = field(default_factory=lambda: False)
+    use_slot_uris: bool = field(default_factory=lambda: False)
+
+
+    def __post_init__(self):
+        self.schemaview = SchemaView(self.schema)
+        super().__post_init__()
+
+    def serialize(self) -> MERMAID_SERIALIZATION:
+        """
+        Serialize a schema as an ER Diagram.
+
+        If a tree_root is present in the schema, then only Entities traversable
+        from here will be included. Otherwise, all Entities will be included.
+
+        :return: mermaid string
+        """
+        sv = self.schemaview
+        structural_roots = [cn for cn, c in sv.all_classes().items() if c.tree_root]
+        if self.structural and structural_roots:
+            return self.serialize_classes(structural_roots, follow_references=True)
+        else:
+            diagram = ERDiagram()
+            for cn in sv.all_classes(imports=False):
```

Comment:
should this be `for cn in sv.all_classes(imports=True):` ?

Reply:
yes, this should probably be the default